src/Sylius/Bundle/CoreBundle/Validator/Constraints/ExistingChannelCodeValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class ExistingChannelCodeValidator extends ConstraintValidator
+{
+    public function __construct(private ChannelRepositoryInterface $channelRepository)
+    {
+    }
+
+    public function validate($value, Constraint $constraint): void
+    {
+        if ($value === null) {
+            return;
+        }
+
+        if ($this->channelRepository->findOneByCode($value) === null) {
+            Assert::isInstanceOf($constraint, ExistingChannelCode::class);
```

Comment:
Should be checked before this `if` and querying the database, probably as the first check in this validator